include/seqan3/alignment/all.hpp

```diff
@@ -10,20 +10,295 @@
+ * inclusive. The front coordinate stores the begin of the alignment part in both sequences.
+ * With the last option the complete alignment is computed. If seqan3::align_cfg::result is not specified only the
+ * score is available. Accessing one of the results that have not been requested will trigger a static assertion
+ * informing the developer about the invalid access.
  *
- * The alignment module contains concepts, functions and classes that are related to the computation of
- * pairwise and multiple sequence alignments.
+ * ## Using scoring and gap schemes
  *
- * \todo Write detailed landing page.
+ * To compute an alignment a scoring and a gap scheme must be provided which give a "score" for substituting, inserting,
+ * or deleting a base within the alignment computation. Throughout SeqAn a positive score implies
+ * higher similarity and/or a closer relatedness and a lower or even negative score implies distance.
+ * If you are used to dealing with "penalties" or "distances", instead think of "negative scores" when using SeqAn
+ * interfaces.
+ *
+ * ### Scoring two letters
+ *
+ * Scoring two letters of a single alphabet (or two similar alphabets) is performed by scoring schemes. A scoring
+ * scheme is any type that models seqan3::ScoringScheme, i.e. it must provide a member function that
+ * takes the two letters and returns the scheme-specific score. Algorithms that expect a scoring scheme should check
+ * this concept with their respective alphabet(s).
+ *
+ * Two generic scoring schemes are provided:
+ *
+ *   1. seqan3::nucleotide_scoring_scheme that accepts all nucleotides (and any alphabet that is explicitly
+ * convertible to seqan3::dna15)
+ *   2. seqan3::aminoacid_scoring_scheme that accepts all amino acids (and any alphabet that is explicitly convertible
+ * to seqan3::aa27).
+ *
+ * These also support scoring two nucleotides/amino acids of different types and they also support modification of
+ * their scores via `set_()` functions and by returning references to their internal score matrix. You can however
+ * add completely different types, as long as they model seqan3::ScoringScheme.
+ *
+ * The scoring scheme can be configured with the seqan3::align_cfg::scoring element. Since the scoring scheme is
+ * strongly coupled on the sequences to be aligned it can not be defaulted. Thus, it is mandatory for the
+ * the developer to specify the seqan3::align_cfg::scoring configuration.
+ *
+ * <small><i>The base type seqan3::scoring_scheme_base is only implementation detail and not required for most users.
+ * Types that model seqan3::ScoringScheme can (but don't need to!) inherit from it.</i></small>
+ *
+ * ### Scoring gaps
+ *
+ * Gaps are usually not scored on a per-character-basis, because it is widely recognised that the likelihood of
+ * `n` consecutive gaps is much higher than that of `n` individual gaps. This is based on the assumption that
```

Comment:
I think we agreed in a core team meeting to use the following terminology:
  * "gap" for "gap symbol" or "individual gap"
  * "gap interval" for "consecutive gaps" or "gap stretch"

Can you
  1. introduce this terminology in this subsection
  2. make sure that you use it yourself everywhere

?